speculation-rules.bs

```diff
@@ -111,10 +121,16 @@ A <dfn>speculation rule</dfn> is a [=struct=] with the following [=struct/items=
 
 The only valid string for [=speculation rule/requirements=] to contain is "`anonymous-client-ip-when-cross-origin`".
 
+The <dfn http-header><code>Speculation-Rules</code></dfn> HTTP response header is a [=structured header=] whose value must be a [=structured header/list=] whose members must be [=structured header/strings=] which are [=valid non-empty URLs potentially surrounded by spaces=].
```

Comment:
I'd probably just move this definition into the "The Speculation-Rules header" section.

I also think this should probably just use ["valid URL string"](https://url.spec.whatwg.org/#valid-url-string) rather than the more flexible "potentially surrounded by space", which I think is exclusively used for stuff parsed straight out of DOM attributes.

Reply:
Done.